### ISSUE TITLE
`CastsAttributes` implementations emit false `MissingTemplateParam` + `MoreSpecificImplementedParamType` even when `@template` binds are correct

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#796]
## Summary

`Illuminate\Contracts\Database\Eloquent\CastsAttributes` is declared with `@template TGet` + `@template TSet`, but implementers hit two stacked false positives:

1. **`MissingTemplateParam`** on every custom cast class — because the PHP-level syntax `implements CastsAttributes` omits `<TGet, TSet>`, even though Laravel's own docs and examples never use template-binding syntax.
2. **`MoreSpecificImplementedParamType`** on `set()` (and sometimes `get()`) — because the interface parameter type in the PHP signature is `mixed`, while user implementations restrict it to the concrete value object type. The `@template TSet` binding is not propagated from phpdoc to actual param-type comparison.

Cumulative effect in real apps: **every custom cast produces 2–3 errors.** Koel has 14 cast classes = 36 errors that are purely stylistic (no runtime issue, LSP is actually respected via the template).

## Environment

- `psalm/plugin-laravel` dev-master @ `8017a1d`
- Psalm 7.0.0-beta19
- Laravel 12 (framework interface unchanged since ~10.x)

## Reproducer

```php
<?php
use Illuminate\Contracts\Database\Eloquent\CastsAttributes;
use Illuminate\Database\Eloquent\Model;

final class LicenseInstance
{
    public function __construct(public string $id) {}
    public function toJson(): string { return json_encode(['id' => $this->id]); }
    public static function fromJson(string $json): self {
        return new self(json_decode($json, true)['id']);
    }
}

class LicenseInstanceCast implements CastsAttributes
//                                   ^^^^^^^^^^^^^^^^
// MissingTemplateParam (psalm/182) - CastsAttributes has missing template params,
//                                    expecting 2
{
    public function get($model, string $key, $value, array $attributes): ?LicenseInstance
    {
        return $value === null ? null : LicenseInstance::fromJson($value);
    }

    public function set($model, string $key, $value, array $attributes): ?string
    //                                             ^
    // MoreSpecificImplementedParamType (psalm/140) — Argument 3 has the more specific
    // type 'LicenseInstance|null', expecting 'mixed' as defined by
    // CastsAttributes::set
    {
        return $value?->toJson();
    }
}
```

Writing `implements CastsAttributes<LicenseInstance, LicenseInstance>` silences (1) but Psalm then emits `MoreSpecificImplementedParamType` on `set()` even harder, because the method signature has `mixed $value` while the phpdoc says `TSet|null`. Binding templates does not reconcile this.

## Root cause

Laravel's interface:

```php
/**
 * @template TGet
 * @template TSet
 */
interface CastsAttributes
{
    /**
     * @param  mixed  $value
     * @return TGet|null
     */
    public function get(Model $model, string $key, mixed $value, array $attributes);

    /**
     * @param  TSet|null  $value
     * @return mixed
     */
    public function set(Model $model, string $key, mixed $value, array $attributes);
}
```

The PHP type-hint on `$value` is `mixed` (cannot be changed without BC break), but the phpdoc uses `TSet|null`. Psalm's LSP check seems to use the PHP-level `mixed` rather than the phpdoc template for variance comparison.

## Suggested fix

The plugin should ship a stub for `Illuminate\Contracts\Database\Eloquent\CastsAttributes` that:

1. Declares the `@template` params defaults so `implements CastsAttributes` (without `<>`) doesn't emit `MissingTemplateParam`. Possible syntax: `@template TGet = mixed` / `@template TSet = mixed`.
2. Uses `@param TSet $value` on the PHP-level signature (or equivalent Psalm-level flag) so subclass narrowing is accepted.

Alternatively, suppress the LSP check entirely for parameters that the interface declares as `mixed` (since any narrowing is intentional).

## Census (koel)

14 cast classes, 36 errors total:
- `app/Casts/EncryptedValueCast.php`
- `app/Casts/LicenseInstanceCast.php`
- `app/Casts/LicenseMetaCast.php`
- `app/Casts/Podcast/CategoriesCast.php`
- `app/Casts/Podcast/EnclosureCast.php`
- `app/Casts/Podcast/EpisodeMetadataCast.php`
- `app/Casts/Podcast/PodcastMetadataCast.php`
- `app/Casts/Podcast/PodcastStateCast.php`
- `app/Casts/SmartPlaylistRulesCast.php`
- `app/Casts/SongLyricsCast.php`
- `app/Casts/SongStorageCast.php`
- `app/Casts/SongTitleCast.php`
- `app/Casts/ThemePropertiesCast.php`
- `app/Casts/UserPreferencesCast.php`

**Comment by alies-dev** [comment]
Converted to the repository internal ideas list (`.alias/issues.md`) so GitHub issues stay focused on active work. Reopen or recreate this when it is prioritized.

**Comment by alies-dev** [comment]
Converted to the repository internal ideas list (`.alias/issues.md`) so GitHub issues stay focused on active work. Reopen or recreate this when it is prioritized.
